pkgs/development/cuda-modules/db/json.nix

```diff
@@ -0,0 +1,276 @@
+            "license_path"
+            "version"
+            "name"
+          ])
+        ) perSystem;
+        result =
+          if hasTags then
+            fmapAttrsToList actualSystems (
+              systemNv: perTag:
+              lib.addErrorContext "While unpacking `${pname}.${systemNv}` as `perTag`" (
+                fmapAttrsToList perTag (
+                  tag: rawPackage:
+                  lib.addErrorContext "While parsing systemNv=${systemNv} tag=${tag} from path=${path}" (
+                    f otherAttrs pname systemNv tag rawPackage
+                  )
+                )
+              )
+            )
+          else
+            fmapAttrsToList actualSystems (systemNv: rawPackage: f otherAttrs pname systemNv null rawPackage);
+      in
+      lib.addErrorContext "While parsing pname=${pname} from path=${path}" result
+    );
+
+  mkDefaultHarder = lib.mkOverride 900;
+  mkDefaultWeaker = lib.mkOverride 1100;
+  pname = lib.mapAttrs (_: _: 1) rawPackages;
+in
+
+{ config, ... }:
+
+{
+  _file = ./json.nix;
+  config = lib.mkMerge (
+    (fmapPackages (
+      otherAttrs: pname: systemNv: _: rawPackage: [
+        {
+          package.name.${pname} =
+            if otherAttrs ? name then mkChooseLonger otherAttrs.name else lib.mkDefault pname;
+        }
+      ]
+    ))
+    ++ (fmapPackages (
+      _: pname: systemNv: _: rawPackage:
+      lib.optionals (looksLikeSystem systemNv) [
+        {
+          package.systemsNv.${pname}.${systemNv} = 1;
+          system.nvidia.${systemNv} = 1;
+        }
+      ]
+    ))
+    ++ lib.optionals (productName != null) [
+      {
+        product.${productName} = 1;
+      }
+    ]
+    ++ lib.optionals (productName != null) (
+      fmapPackages (
+        otherAttrs: pname: systemNv: tag: rawPackage:
+        lib.optionals (looksLikeSystem systemNv && otherAttrs ? version) [
+          {
+            release.${productName}.${pname} = {
+              ${releaseLabel} = otherAttrs.version;
+            };
+            archive.bucket.${pname}.${otherAttrs.version}.${rawPackage.sha256} = 1;
+            archive.sha256.${rawPackage.sha256} =
+              {
+                inherit systemNv;
+                url = "${config.base_url}${rawPackage.relative_path}";
+              }
+              // lib.optionalAttrs (tag != null) {
+                tags.${tag} = 1;
+              };
+          }
+        ]
+      )
+    )
+    ++ [
+      {
+        package = {
+          inherit pname;
+          license =
+            let
+              licenseWithPriority =
+                otherAttrs: pname: systemNv: _: rawPackage:
+                let
+                  license = licenseOf pname otherAttrs;
+                in
+                lib.optionals (license != null) [
+                  {
+                    ${pname} = mkChooseLonger license;
+                  }
+                ];
+            in
+            lib.mergeAttrsList (fmapPackages licenseWithPriority);
```

Comment:
Can you rewrite this so we're not doing explicit merging, and instead allow the module system to do all of it for us?

Reply:
Originally I was trying to avoid using `mergeAttrsList` too. My hunch is that merging a few big definitions is cheaper than multiple smaller definitions when it comes to `evalModules`